src/rulesets/oas/functions/__tests__/typedEnum.test.ts

```diff
@@ -1,9 +1,11 @@
+import { functions } from '../../../../functions';
 import { typedEnum } from '../typedEnum';
 
 const defaultReportingThreshold = 3;
 
 function runTypedEnum(targetVal: any, reportingThreshold: any) {
-  return typedEnum(
+  return typedEnum.call(
```

Comment:
@P0lip Same question here. When should this invocation method be favored?

Reply:
Everything `setFunctionContext` does is setting the "`this` context".
It uses `Function.prototype.bind` underneath.
That being said, you can either use `setFunctionContext` or Function.prototype.{bind,call,apply}, whatever works for you the best.